cmake/SetupCompiler.cmake

```diff
@@ -62,6 +62,14 @@ elseif(CMAKE_CXX_COMPILER_ID STREQUAL "Clang" OR CMAKE_CXX_COMPILER_ID STREQUAL
         set(xad_cxx_extra -mavx2)
     elseif(XAD_SIMD_OPTION STREQUAL AVX512)
         set(xad_cxx_extra -march=cascadelake)
+    elseif(XAD_SIMD_OPTION STREQUAL NATIVE)
+        set(xad_cxx_extra  -march=native)
+    elseif(XAD_SIMD_OPTION STREQUAL APPLE_M1)
+        if(CMAKE_CXX_COMPILER_ID STREQUAL "Clang")
+            set(xad_cxx_extra  -march=apple-m1)
+        elseif(CMAKE_CXX_COMPILER_ID STREQUAL "GNU")
```

Comment:
note that this is in the branch for Clang or AppleClang - the compiler cannot be `GNU` here.